### PR TITLE
Add docstrings for `Dialect`s, update README

### PR DESCRIPTION
**Pull request by alamb**
I want to point out that dialect specific syntax should be supported for both `GenericDialect` and the specific `Dialect`

I also took the time to update the doc strings and the main README

### CODE REVIEW

**Comment by alamb** [comment]
@Dandandan  / @andygrove  if you have time to review this PR I would appreciate it

**Comment by alamb** [comment]
cc @lovasoa  for your thoughts

File: README.md [hunk 1]
```diff
@@ -124,28 +124,36 @@ parser](docs/custom_sql_parser.md).
...
-maintain this crate is fairly limited.
+maintain this crate is limited. Please read the following sections carefully.
 
-Pull requests that add support for or fix a bug in a feature in the
-SQL standard, or a feature in a popular RDBMS, like Microsoft SQL
+### New Syntax
+
+The most commonly accepted PRs add support for or fix a bug in a feature in the
+SQL standard, or a a popular RDBMS, such as Microsoft SQL
 Server or PostgreSQL, will likely be accepted after a brief
```
Comment (andygrove) [inline README.md:133]:
```suggestion
Server or PostgreSQL, and will likely be accepted after a brief
```

File: README.md [hunk 2]
```diff
@@ -124,28 +124,36 @@ parser](docs/custom_sql_parser.md).
...
-review.
+review.  Any SQL feature that is dialect specific should be parsed by *both* the relevant [`Dialect`] 
+as well as the omnibsus [`GenericDialect`].
```
Comment (andygrove) [inline README.md:135]:
```suggestion
as well as the omnibus [`GenericDialect`].
```

Comment (andygrove) [reply README.md:135]: This word may be confusing to non-native English speakers.

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @alamb

**Review by lovasoa — APPROVED** [review]
:+1:

File: README.md (hunk 2 above)
Comment (alamb) [reply README.md:135]: Good call -- I think it is not necessary in this context. 

```suggestion
as well as [`GenericDialect`].
```

**Comment by alamb** [comment]
Thank you @alamb  and @lovasoa

**Pull request merged**